### PR TITLE
Implement filter in dynamo

### PR DESCRIPTION
**Pull request by isuruf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131674

Fixes https://github.com/pytorch/pytorch/issues/128944

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Comment by xinyu-intel** [comment]
@isuruf Hi, can this feature be merged?

**Comment by isuruf** [comment]
This needs a review. Can you review?

cc @amjames

**Review by amjames — APPROVED** [review]

File: torch/_dynamo/variables/builtin.py [hunk 1]
```diff
@@ -1518,6 +1518,20 @@ def call_map(self, tx: "InstructionTranslator", fn, *seqs):
...
+            try:
+                items = list(
+                    filter(
+                        lambda x: fn.call_function(tx, [x], {}).as_python_constant(),
+                        seq_unpacked,
+                    )
+                )
+                return variables.TupleVariable(items)
+            except NotImplementedError:
+                return
```
Comment (jansel) [inline torch/_dynamo/variables/builtin.py:1533]: Can you add a test the exercise this code path?

**Review by jansel — CHANGES_REQUESTED** [review]

**Review by jansel — APPROVED** [review]

**Comment by tmct** [comment]
Thank you for fix @isuruf - do you know roughly when this might get released? It doesn't look like it made it into 2.4.1

**Comment by isuruf** [comment]
I assume this will be in the 2.5.0 release.

**Pull request closed without merging**